### PR TITLE
North schedules with stats info [FOGL-1635]

### PR DESCRIPTION
**Pull request by praveen-garg**
Depends on FOGL-1640. Once FOGL-1640 stuff is merged into develop, this shall need a rebase.

### CODE REVIEW

File: python/foglamp/services/core/routes.py [hunk 1]
```diff
@@ -91,6 +92,9 @@ def setup(app):
     app.router.add_route('POST', '/foglamp/service', service.add_service)
     app.router.add_route('GET', '/foglamp/service', service.get_health)
 
+    # North
+    app.router.add_route('GET', '/foglamp/north', north.get_north_schedules)
```
Comment (praveen-garg) [inline python/foglamp/services/core/routes.py]: suggest a better name!

File: python/foglamp/services/core/api/north.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        stats = await _get_sent_stats()
+
+        for sch in north_schedules:
+            # FIXME: sch.name vs sch.processName?
+            stat = next((s for s in stats if s["key"] == sch["name"]), None)
+            if stat:
+                sch["sent"] = stat["value"]
+            else:
+                # FIXME: what if no key && schedule match in statistics found
+                sch["sent"] = -1
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/north.py]: what if a disabled schedule exists and no stats created yet; so 0 as default or -1

File: python/foglamp/services/core/api/north.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    :Example:
+            curl -X GET http://localhost:8081/foglamp/north
+    """
+    try:
+        north_schedules = await _get_north_schedules()
+        stats = await _get_sent_stats()
+
+        for sch in north_schedules:
+            # FIXME: sch.name vs sch.processName?
+            stat = next((s for s in stats if s["key"] == sch["name"]), None)
```
Comment (amarendra-dianomic) [inline python/foglamp/services/core/api/north.py]: Safe to use sch["processName"], with a TODO, for now as this is due to a bug in the scheduler code which will be removed in a near future sprint.

File: python/foglamp/services/core/api/north.py (hunk 2 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/core/api/north.py]: IMHO, 0 is more intuitive as we also provide a "enabled" col.

**Comment by praveen-garg** [comment]
Agree @MarkRiddoch ! these shall really need better and thoughtful treatment, Hence just 2 seprate endpoint route and 2 module files.

File: python/foglamp/services/core/api/north.py (hunk 2 above)
Comment (praveen-garg) [reply python/foglamp/services/core/api/north.py]: gui shall only show if `>=0`

**Pull request merged**